### ISSUE TITLE
⌨ Focus not actually trapped in mobile menu

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#13143]
## ⌨ Accessibility Issue

When using the mobile menu, focus is trapped as expected. However, for some reason I can leave the trap when I have a screen reader enabled.

## 🦋 Expected Behaviour

Whe the mobile menu is open, the focus cannot leave the menu to highlight items obscurred by it. if we cannot do that, we should at least close the menu when focus leaves as to not confuse users.

## 🕵️ Details

- **Operating System**: MacOS
- **Browser**: Chrome/Safari
- **Assistive Technology**: Voiceover

## 📋 Steps to Reproduce

> [!IMPORTANT]
> Honestly, this is pretty difficult to reproduce. I was lucky enough to get a screen recording after trying very hard to get my focus to move on from the home button 😓  
> 
> Actually, I am able to reliably reproduce this on android + talkback. Screen recording added

1. Navigate to `talent.canada.ca`
2. Turn on voiceover
3. Open the mobile menu
4. Navigate through the links
5. After a while somehow you can break out of the trap??? 😕 

## 📸 Screenshot

https://github.com/user-attachments/assets/b895e570-7c77-4921-9a13-41c41550250e

https://github.com/user-attachments/assets/948fb7eb-1bed-46d9-bade-42e03f3624a9

## 🙋‍♀️ Proposed Solution

Really no idea, maybe the third party focus trap has a bug? Maybe we just need to use radix?

## ✅ Acceptance Criteria

- [ ] Mobile menu actually traps your focus

**Comment by mnigh** [comment]
I am able to reproduce this on macOS 12.7.6 using Safari 17.6 and Chrome 135.
